nailgun/entities.py

```diff
@@ -1463,14 +1473,23 @@ def __init__(self, server_config=None, **kwargs):
                 required=True,
             ),
         }
+        super(HostGroupClasses, self).__init__(server_config, **kwargs)
         self._meta = {
-            'api_path': 'api/v2/hostgroups/:hostgroup_id/puppetclass_ids',
             'server_modes': ('sat'),
+            'api_path': '{0}/puppetclass_ids'.format(
+                # pylint:disable=no-member
+                self.hostgroup.path('self')
+            )
         }
-        super(HostGroupClasses, self).__init__(server_config, **kwargs)
+
+    def read(self, entity=None, attrs=None, ignore=('hostgroup',)):
+        """Deal with redundant entity fields"""
+        if attrs is None:
+            attrs = self.read_json()
```

Comment:
Completely re-implemented. Thanks for pointing into right direction.
It is best solution that I find after some hours of debugging